code/game/objects/items/weapons/lighters.dm

```diff
@@ -19,10 +19,15 @@
 	/// Cooldown until the next turned off message/sound can be activated
 	var/next_off_message
 
-/obj/item/lighter/random/New()
+/obj/item/lighter/random
+	/// Random color that is assigned whenever we initialize a new random color lighter.
+	var/random_color = "g"
```

Comment:
Im pretty sure we've removed this in a previous PR. Maybe you could try something like `base_icon_state` instead?

Reply:
We didn't, the PR got closed
You can take a look at #23132 for a great example on how to fix this!